packages/typescript/src/internal/build_defs.bzl

```diff
@@ -284,7 +284,11 @@ def _ts_library_impl(ctx):
         ),
         # TODO: Add remaining shared JS provider from design doc
         # (JSModuleInfo) and remove legacy "typescript" provider
-        # once it is no longer needed.
+        # once it is no longer needed.,
+
+        # indicates that the files listed in this invocation under srcs should be instrumented by coverage
+        # also transitively collect from deps
+        coverage_common.instrumented_files_info(ctx, source_attributes = ["srcs"], dependency_attributes = ["deps"]),
```

Comment:
Will need this in ng_module and any other rules that transform source files to `.js` looks like?

Reply:
Yup! any rule that provides fules which should be instrumented will need this.
I don't know how it'll work for pure JS files, any ideas there?

Reply:
That's a good point. js_library may be useful in this regard. Also, in my pkg_npm PR I'm able to remove the `module_from_src` which was previously required by the linker as the rule can determine if the srcs are source files or generated files and ensure they are all uniform. Provides a lightweight linkable package with `module_name`.... although`module_name` is a bad name and should be renamed to `pacakge_name` as it is named in `pkg_npm` in that PR.

Reply:
https://github.com/bazelbuild/rules_nodejs/pull/1715/files#diff-850cec1f6d2867d724158a5607f79fee

Reply:
I removed this line to see how it would affect the coverage output and it was not affected. Only removing the `instrumented_files_info ` from `node.bzl` broke the coverage with
```
test/coverage_test.sh: line 98: COVERAGE_DIR: unbound variable
```
assuming on the line generated by
```
env_vars += "export NODE_V8_COVERAGE=$COVERAGE_DIR\n"
```